gateway-js/src/config.ts

```diff
@@ -119,71 +124,181 @@ interface GatewayConfigBase {
 
   // experimental observability callbacks
   experimental_didResolveQueryPlan?: Experimental_DidResolveQueryPlanCallback;
-  experimental_didFailComposition?: Experimental_DidFailCompositionCallback;
-  experimental_didUpdateComposition?: Experimental_DidUpdateCompositionCallback;
+  didUpdateSupergraph?: DidUpdateSupergraphCallback;
```

Comment:
My instincts say that this is pretty close to onSchemaLoadOrUpdate and it's a little funky to make this thing non-experimental when it is so similar but works differently for non-deep reasons (one top-level config vs a method you can call to register a listener, thrown errors are not caught).

Not gonna block approval on this, happy to go any way, but my personal instincts would be to either:
- leave as experimental
- add its functionality to onSchemaLoadOrUpdate and drop the experimental one

Reply:
Agreed, updated to keep it experimental. Its move to non-experimental status should probably be decided on by not just me (and we might decide its functionality should converge with listeners)